### PR TITLE
feat: support creation of WlRegion, used for setting input region

### PR DESCRIPTION
**Pull request by nairaner**
I wanted to create overlay "transparent"  for mouse/input events, but wasn't able to create WlRegion.

### CODE REVIEW

File: layershellev/examples/simplelayer.rs [hunk 1]
```diff
@@ -23,6 +23,15 @@ fn main() {
             // NOTE: this will send when init, you can request bind extra object from here
             LayerEvent::InitRequest => ReturnData::RequestBind,
             LayerEvent::BindProvide(globals, qh) => {
+                // Example: pass through input events, useful for overlays
+                // use ::wayland_client::protocol::wl_compositor;
+                // if let Some(compositor) = globals.bind::<wl_compositor::WlCompositor, _, _>(&qh, 1..=5, ()).ok() {
+                //     if let Some(state) = ev.get_unit_with_id(index.unwrap()) {
```
Comment (Decodetalkers) [inline layershellev/examples/simplelayer.rs:29]: Wl_compositer is already in the `running`, why not add another returnvalue for it?

File: layershellev/src/lib.rs [hunk 2]
```diff
@@ -1387,6 +1388,18 @@ impl<T> Dispatch<xdg_surface::XdgSurface, ()> for WindowState<T> {
         }
     }
 }
+
+impl<T> Dispatch<WlRegion, ()> for WindowState<T> {
```
Comment (Decodetalkers) [inline layershellev/src/lib.rs:1392]: I think it can use dispatch_noop! macro here

**Comment by Decodetalkers** [comment]
Thank you very much!

File: layershellev/src/lib.rs (hunk 2 above)
Comment (nairaner) [reply layershellev/src/lib.rs]: Thanks, I didn't understand how that worked, it seems better

File: layershellev/src/events.rs [hunk 3]
```diff
@@ -121,6 +126,7 @@ pub enum ReturnData<INFO> {
     WlBuffer(WlBuffer),
     RequestBind,
     RequestExist,
```
Comment (nairaner) [inline layershellev/src/events.rs:128]: I think this should be called `RequestExit` ?

**Comment by nairaner** [comment]
Thanks for feedback. I didn't understand code well enough to figure out how to connect all components. Now it should be better.

File: layershellev/src/events.rs (hunk 3 above)
Comment (Decodetalkers) [reply layershellev/src/events.rs:128]: Yes, it is a typo, you can help fix it for me

File: layershellev/src/lib.rs [hunk 4]
```diff
@@ -1824,6 +1826,13 @@ impl<T: 'static> WindowState<T> {
                         None,
                     ));
                 }
+                Some(ReturnData::RequestCompositor) => {
```
Comment (Decodetalkers) [inline layershellev/src/lib.rs:1829]: Emm. I mean add another return value like `RequestSetRegion`... and it should not be in this place I think

**Comment by Decodetalkers** [comment]
Emm. must the region be set only once time? if just once, I think this pr is ok

**Comment by Decodetalkers** [comment]
ok. I will merge it now. thank you for your effort

**Pull request merged**